packages/hash/shared/src/types.ts

```diff
@@ -1,8 +1,46 @@
-// The types exported here are expected to be branded.
+import {
+  EntityId,
+  extractEntityUuidFromEntityId as extractEntityUuidFromEntityIdSubGraph,
+  extractOwnedByIdFromEntityId as extractOwnedByIdFromEntityIdSubGraph,
+  splitEntityId as splitEntityIdSubGraph,
+} from "@hashintel/hash-subgraph";
 
-export type Uuid = string;
-export type AccountId = Uuid;
-export type OwnedById = Uuid;
-export type UpdatedById = Uuid;
+type Branded<Base, Kind extends string> = {
+  /** The type of the value space that is branded */
+  readonly _base: Base;
+  /** The unique name for the branded type. */
+  readonly _kind: Kind;
+};
 
-export type EntityUuid = Uuid;
+/**
+ * The type-branding type to support nominal (name based) types.
+ */
+export type Brand<Base, Kind extends string> = Base & Branded<Base, Kind>;
+
+/** An account ID of an actor that is the owner of something */
+export type OwnedById = Brand<string, "OwnedById">;
+/** An account ID of an actor that has updated something */
+export type UpdatedById = Brand<string, "UpdatedById">;
+/** An ID to uniquely identify an account (e.g. a User or an Org) */
+export type AccountId = Brand<string, "AccountId"> | OwnedById | UpdatedById;
```

Comment:
I think we probably want an `ActorId` alongside these (inheriting from `AccountId`)

Reply:
Hmm, I see why we may want that, but I have a feeling it may be equivalent to `UpdatedById` in terms of how useful it is. We wouldn't be returning `ActorId` from anything, only taking it as a param. If it's inheriting from `AccountId` it would require an explicit upcast to be used as an argument